File: bank_backend/src/main/java/com/bank/sbnz/model/EmploymentInfo.java
package com.bank.sbnz.model;


import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class EmploymentInfo {
    private boolean employed;
    private boolean indefinitely;
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonDeserialize(using = LocalDateDeserializer.class)
    private LocalDate employmentStartDate;
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonDeserialize(using = LocalDateDeserializer.class)
    private LocalDate employmentEndDate;
    private double monthlyIncome;
}

File: bank_frontend/src/app/modules/bookstore/model/employmentInfo.ts
export class EmploymentInfo {
    employed: boolean = false;
    indefinitely: boolean = false;
    employmentStartDate: string = '';
    employmentEndDate: string = '';
    monthlyIncome: number = 0;
  
    public constructor(obj?: any) {
      if(obj) {
        this.employed = obj.employed;
        this.indefinitely = obj.indefinitely;
        this.employmentStartDate = obj.employmentStartDate;
        this.employmentEndDate = obj.employmentEndDate;
        this.monthlyIncome = obj.monthlyIncome;
      }
    }
  }